src/routes/+layout.server.ts

```diff
@@ -46,25 +46,27 @@ export const load: LayoutServerLoad = async ({ locals, depends }) => {
 		});
 	}
 
-	// get the number of messages where `from === "assistant"` across all conversations.
-	const totalMessages =
-		(
-			await collections.conversations
-				.aggregate([
-					{ $match: authCondition(locals) },
-					{ $project: { messages: 1 } },
-					{ $unwind: "$messages" },
-					{ $match: { "messages.from": "assistant" } },
-					{ $count: "messages" },
-				])
-				.toArray()
-		)[0]?.messages ?? 0;
-
 	const messagesBeforeLogin = MESSAGES_BEFORE_LOGIN ? parseInt(MESSAGES_BEFORE_LOGIN) : 0;
 
-	const userHasExceededMessages = messagesBeforeLogin > 0 && totalMessages > messagesBeforeLogin;
-
-	const loginRequired = requiresUser && !locals.user && userHasExceededMessages;
+	let loginRequired = false;
+
+	if (requiresUser && !locals.user && messagesBeforeLogin) {
+		// get the number of messages where `from === "assistant"` across all conversations.
+		const totalMessages =
+			(
+				await collections.conversations
+					.aggregate([
```

Comment:
overall, `aggregate` is an expensive operation

Reply:
Nice, hopefully this will help quite a bit! Can we also reduce the number of times where this is needed even more by first checking the conversations count ?

For example if the user has more than `messagesBeforeLogin` conversations, then there shouldn't be a need to aggregate at all, wdyt? Not entirely sure about it

Reply:
handled in https://github.com/huggingface/chat-ui/pull/863/commits/fe99cf2dc9d89c6ba91c8fc096a273cc317de23e